script, test: python linter flake8 E275 fixup, update dependencies

jonatack opened the pull request:
It is helpful to be able to run the python linter locally to review PRs and check local diffs and work.  Fix the errors raised by `./test/lint/lint-python.py` when run locally with flake8 5.0.4, which enforces rule E275 more strictly than previous versions, and update our python linter CI dependencies.

jonatack committed:
script, test: fix python linter E275 errors with flake8 5.0.4

jonatack committed:
script: update python linter dependencies

stickies-v reviewed (commented):
Approach ACK - this has been on my to-do list too. I prefer fixing the issue as opposed to dropping rules.

stickies-v reviewed (approved):
ACK 395519be6

Nice to have `lint-python.py` run without all those warnings. Non-controversial change imo, could have put a space instead of removing the parenthesis but this is more pythonic so I'm in favour. We do have instances of `assert (...)` left where the parentheses _could_ be removed but it's orthogonal and not worth the change on its own, imo.

brunoerg reviewed (commented):
Concept ACK

w0xlt reviewed (approved):
ACK https://github.com/bitcoin/bitcoin/pull/26257/commits/395519be61090cd9e8ae1f6d48399a3fb99ca9ec

MarcoFalke commented:
CI is red

MarcoFalke commented:
Also, not sure if we should enforce this whitespace, given that no other whitespace is enforced.

jonatack commented:
Rebased and updated with latest fixups (CI was green on the last push with 2 ACKs here, so I suppose it was restarted to verify with the latest merges).

jonatack commented:
> Also, not sure if we should enforce this

As mentioned in the pull description, another option would be to drop the E275 rule.

MarcoFalke commented:
CI is still red.

To me it seems odd to enforce this whitespace but no other. If you want to enforce whitespace, it might be best to discuss in a separate pull, and then for all whitespace, unless there is a reason to do it for only this one.

jonatack commented:
Rebased and CI is green. The intent is only to have our python linter pass on CI and locally with the current version; E275 seems to be part of current flake8 (https://www.flake8rules.com/rules/E275.html) unless we opt out of it. Like @stickies-v, I mildly prefer fixing the issue with more pythonic code rather than dropping rules.

fanquake commented:
~0

Is there a particular reason this change is broken into commits per-directory? Seems like an odd way to break it up.
Could squash the 4, and use a commit message more explicit than "fixups" given there's only a single change here?

jonatack commented:
> See also [634cb38](https://github.com/bitcoin/bitcoin/commit/634cb384347463d33417ba7b43b189240c8883f1) from #26468, that uses a scripted-diff, which seems like the best way to do this.

I find that version less pythonic and the other comments are nits. It would be nice to fix this and this has been open for a while.

fanquake commented:
> and the other comments are nits.

I don't think suggesting the use of a commit message that clearly explains a change, rather than multiple vauge messages, is a nit.

The message from https://github.com/bitcoin/bitcoin/commit/634cb384347463d33417ba7b43b189240c8883f1: `"Fix E275 errors in lint-python.py with flake8 5.0.4"`, explains what's being done, and why. This is much more useful when compared to `"X: python linter X fixups"` repeated 4 times.

willcl-ark reviewed (approved):
ACK fc3ca6f87f16448c4f4b9ecc3afa813f12868c18

Happy to re-review if you do change up the commit messages.

I do prefer the result of this PR over how my scripted diff turned it out (sorry again @jonatack for opening that dupe without searching first!)

stickies-v reviewed (approved):
re-ACK https://github.com/bitcoin/bitcoin/commit/fc3ca6f87f16448c4f4b9ecc3afa813f12868c18

However, I think fanquake's suggestion to make the commit messages more helpful is on point, so I would support doing that first. I hereby commit to quickly re-ACK once that's done (assuming no other big changes).

brunoerg reviewed (approved):
ACK fc3ca6f87f16448c4f4b9ecc3afa813f12868c18

Tested locally the versions modified in `04_install` and reviewed the changes in the code, most are similar. I agree on squashing and making the message more helpful, happy to re-ack if you address it!

jonatack commented:
Rebased and addressed feedback. Opened #26801 as an alternative. I mildly prefer this version for its more pythonic code.

kristapsk reviewed (approved):
ACK 1e5e87cec32260a6dfd03f11040aed8b7bb4064e

kristapsk commented:
> I mildly prefer this version for its more pythonic code.

I also prefer this over #26801.
